### ISSUE TITLE
[Bug]: The query result is empty after Milvus recovers from a querynode pod kill or pod failure chaos

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#32449]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master-20240418-18c3cbe4-amd64
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

consistency_level is strong

```
[2024-04-18T18:03:13.129Z] ___________ TestAllCollection.test_milvus_default[Checker__vaAlMs3o] ___________

[2024-04-18T18:03:13.129Z] [gw2] linux -- Python 3.8.10 /usr/bin/python3.8

[2024-04-18T18:03:13.129Z] 

[2024-04-18T18:03:13.129Z] self = <test_all_collections_after_chaos.TestAllCollection object at 0x7ff0f5a3e970>

[2024-04-18T18:03:13.129Z] collection_name = 'Checker__vaAlMs3o'

[2024-04-18T18:03:13.129Z] 

[2024-04-18T18:03:13.129Z]     @pytest.mark.tags(CaseLabel.L1)

[2024-04-18T18:03:13.129Z]     def test_milvus_default(self, collection_name):

[2024-04-18T18:03:13.129Z]         self._connect()

[2024-04-18T18:03:13.129Z]         # create

[2024-04-18T18:03:13.129Z]         name = collection_name if collection_name else cf.gen_unique_str("Checker_")

[2024-04-18T18:03:13.129Z]         t0 = time.time()

[2024-04-18T18:03:13.129Z]         schema = Collection(name=name).schema

[2024-04-18T18:03:13.129Z]         collection_w = self.init_collection_wrap(name=name, schema=schema)

[2024-04-18T18:03:13.129Z]         tt = time.time() - t0

[2024-04-18T18:03:13.129Z]         assert collection_w.name == name

[2024-04-18T18:03:13.129Z]         # get collection info

[2024-04-18T18:03:13.129Z]         schema = collection_w.schema

[2024-04-18T18:03:13.129Z]         dim = cf.get_dim_by_schema(schema=schema)

[2024-04-18T18:03:13.129Z]         int64_field_name = cf.get_int64_field_name(schema=schema)

[2024-04-18T18:03:13.129Z]         float_vector_field_name = cf.get_float_vec_field_name(schema=schema)

[2024-04-18T18:03:13.129Z]         float_vector_field_name_list = cf.get_float_vec_field_name_list(schema=schema)

[2024-04-18T18:03:13.129Z]         # compact collection before getting num_entities

[2024-04-18T18:03:13.129Z]         collection_w.flush(timeout=180)

[2024-04-18T18:03:13.129Z]         collection_w.compact()

[2024-04-18T18:03:13.129Z]         collection_w.wait_for_compaction_completed(timeout=720)

[2024-04-18T18:03:13.129Z]     

[2024-04-18T18:03:13.129Z]         entities = collection_w.num_entities

[2024-04-18T18:03:13.129Z]         log.info(f"assert create collection: {tt}, init_entities: {entities}")

[2024-04-18T18:03:13.129Z]     

[2024-04-18T18:03:13.129Z]         # insert

[2024-04-18T18:03:13.129Z]         offset = -3000

[2024-04-18T18:03:13.129Z]         data = cf.get_column_data_by_schema(nb=ct.default_nb, schema=schema, start=offset)

[2024-04-18T18:03:13.129Z]         t0 = time.time()

[2024-04-18T18:03:13.129Z]         _, res = collection_w.insert(data)

[2024-04-18T18:03:13.129Z]         tt = time.time() - t0

[2024-04-18T18:03:13.129Z]         log.info(f"assert insert: {tt}")

[2024-04-18T18:03:13.129Z]         assert res

[2024-04-18T18:03:13.129Z]     

[2024-04-18T18:03:13.129Z]         # flush

[2024-04-18T18:03:13.129Z]         t0 = time.time()

[2024-04-18T18:03:13.129Z]         _, check_result = collection_w.flush(timeout=180)

[2024-04-18T18:03:13.129Z]         assert check_result

[2024-04-18T18:03:13.129Z]         # assert collection_w.num_entities == len(data[0]) + entities

[2024-04-18T18:03:13.129Z]         tt = time.time() - t0

[2024-04-18T18:03:13.129Z]         entities = collection_w.num_entities

[2024-04-18T18:03:13.129Z]         log.info(f"assert flush: {tt}, entities: {entities}")

[2024-04-18T18:03:13.129Z]     

[2024-04-18T18:03:13.129Z]         # show index infos

[2024-04-18T18:03:13.129Z]         index_infos = [index.to_dict() for index in collection_w.indexes]

[2024-04-18T18:03:13.129Z]         log.info(f"index info: {index_infos}")

[2024-04-18T18:03:13.129Z]         fields_created_index = [index["field"] for index in index_infos]

[2024-04-18T18:03:13.129Z]     

[2024-04-18T18:03:13.129Z]         # create index if not have

[2024-04-18T18:03:13.129Z]         index_params = {"index_type": "HNSW", "metric_type": "L2", "params": {"M": 48, "efConstruction": 500}}

[2024-04-18T18:03:13.129Z]     

[2024-04-18T18:03:13.129Z]         for f in float_vector_field_name_list:

[2024-04-18T18:03:13.129Z]             if f not in fields_created_index:

[2024-04-18T18:03:13.129Z]                 t0 = time.time()

[2024-04-18T18:03:13.129Z]                 index, _ = collection_w.create_index(field_name=float_vector_field_name,

[2024-04-18T18:03:13.129Z]                                                      index_params=index_params)

[2024-04-18T18:03:13.129Z]                 tt = time.time() - t0

[2024-04-18T18:03:13.129Z]                 log.info(f"create index for field {f} cost: {tt} seconds")

[2024-04-18T18:03:13.129Z]         # show index infos

[2024-04-18T18:03:13.129Z]         index_infos = [index.to_dict() for index in collection_w.indexes]

[2024-04-18T18:03:13.129Z]         log.info(f"index info: {index_infos}")

[2024-04-18T18:03:13.129Z]     

[2024-04-18T18:03:13.129Z]         # load

[2024-04-18T18:03:13.129Z]         collection_w.load()

[2024-04-18T18:03:13.129Z]     

[2024-04-18T18:03:13.129Z]         # search

[2024-04-18T18:03:13.129Z]         search_vectors = cf.gen_vectors(1, dim)

[2024-04-18T18:03:13.129Z]         search_params = {"metric_type": "L2", "params": {"ef": 64}}

[2024-04-18T18:03:13.129Z]         t0 = time.time()

[2024-04-18T18:03:13.129Z]         res_1, _ = collection_w.search(data=search_vectors,

[2024-04-18T18:03:13.129Z]                                        anns_field=float_vector_field_name,

[2024-04-18T18:03:13.129Z]                                        param=search_params, limit=1)

[2024-04-18T18:03:13.129Z]         tt = time.time() - t0

[2024-04-18T18:03:13.129Z]         log.info(f"assert search: {tt}")

[2024-04-18T18:03:13.129Z]         assert len(res_1) == 1

[2024-04-18T18:03:13.129Z]         # query

[2024-04-18T18:03:13.129Z]         term_expr = f'{int64_field_name} in {[i for i in range(offset, 0)]}'

[2024-04-18T18:03:13.129Z]         t0 = time.time()

[2024-04-18T18:03:13.129Z]         res, _ = collection_w.query(term_expr)

[2024-04-18T18:03:13.129Z]         tt = time.time() - t0

[2024-04-18T18:03:13.129Z]         log.info(f"assert query result {len(res)}: {tt}")

[2024-04-18T18:03:13.129Z] >       assert len(res) >= len(data[0])

[2024-04-18T18:03:13.129Z] E       assert 0 >= 2000

[2024-04-18T18:03:13.129Z] E        +  where 0 = len([])

[2024-04-18T18:03:13.129Z] E        +  and   2000 = len([-3000, -2999, -2998, -2997, -2996, -2995, ...])

[2024-04-18T18:03:13.129Z] 

[2024-04-18T18:03:13.129Z] testcases/test_all_collections_after_chaos.py:107: AssertionError

[2024-04-18T18:03:13.129Z] ------------------------------ Captured log setup ------------------------------

[2024-04-18T18:03:13.129Z] [2024-04-18 17:54:44 - INFO - ci_test]: *********************************** setup *********************************** (client_base.py:44)

[2024-04-18T18:03:13.129Z] [2024-04-18 17:54:44 - INFO - ci_test]: [setup_method] Start setup test case test_milvus_default. (client_base.py:45)

[2024-04-18T18:03:13.129Z] ------------------------------ Captured log call -------------------------------

[2024-04-18T18:03:13.129Z] [2024-04-18 17:54:44 - DEBUG - ci_test]: (api_request)  : [Connections.connect] args: ['default', '', '', 'default', ''], kwargs: {'host': '10.255.230.215', 'port': 19530} (api_request.py:62)

[2024-04-18T18:03:13.129Z] [2024-04-18 17:54:44 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-04-18T18:03:13.129Z] [2024-04-18 17:54:44 - DEBUG - ci_test]: (api_request)  : [Connections.has_connection] args: ['default'], kwargs: {} (api_request.py:62)

[2024-04-18T18:03:13.129Z] [2024-04-18 17:54:44 - DEBUG - ci_test]: (api_response) : True  (api_request.py:37)

[2024-04-18T18:03:13.129Z] [2024-04-18 17:54:44 - DEBUG - ci_test]: (api_request)  : [Collection] args: ['Checker__vaAlMs3o', {'auto_id': False, 'description': '', 'fields': [{'name': 'int64', 'description': '', 'type': <DataType.INT64: 5>, 'is_primary': True, 'auto_id': False}, {'name': 'float', 'description': '', 'type': <DataType.FLOAT: 10>}, {'name': 'varchar', 'description': '', 'type': <DataType......, kwargs: {'consistency_level': 'Strong'} (api_request.py:62)

[2024-04-18T18:03:13.129Z] [2024-04-18 17:54:44 - DEBUG - ci_test]: (api_response) : <Collection>:

[2024-04-18T18:03:13.129Z] -------------

[2024-04-18T18:03:13.130Z] <name>: Checker__vaAlMs3o

[2024-04-18T18:03:13.130Z] <description>: 

[2024-04-18T18:03:13.130Z] <schema>: {'auto_id': False, 'description': '', 'fields': [{'name': 'int64', 'description': '', 'type': <DataType.INT64: 5>, 'is_primary': True, 'auto_id': False}, {'name': 'float', 'description': '', 'type': <DataType.FLOAT: 10>}......  (api_request.py:37)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:54:44 - DEBUG - ci_test]: (api_request)  : [Collection.flush] args: [], kwargs: {'timeout': 180} (api_request.py:62)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:54:44 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:54:44 - DEBUG - ci_test]: (api_request)  : [Collection.compact] args: [180], kwargs: {} (api_request.py:62)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:54:44 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:58:35 - DEBUG - ci_test]: (api_request)  : [Collection.flush] args: [], kwargs: {'timeout': 180} (api_request.py:62)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:58:36 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:58:36 - INFO - ci_test]: assert create collection: 0.016424894332885742, init_entities: 430421 (test_all_collections_after_chaos.py:49)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:58:39 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [[[-3000, -2999, -2998, -2997, -2996, -2995, -2994, -2993, -2992, -2991, -2990, -2989, -2988, -2987, -2986, -2985, -2984, -2983, -2982, -2981, -2980, -2979, -2978, -2977, -2976, -2975, -2974, -2973, -2972, -2971, -2970, -2969, -2968, -2967, -2966, -2965, -2964, -2963, -2962, -2961, -2960, -2959, -29......, kwargs: {'timeout': 180} (api_request.py:62)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:58:40 - DEBUG - ci_test]: (api_response) : (insert count: 2000, delete count: 0, upsert count: 0, timestamp: 449174076300984323, success count: 2000, err count: 0)  (api_request.py:37)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:58:40 - INFO - ci_test]: assert insert: 1.7275185585021973 (test_all_collections_after_chaos.py:57)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:58:40 - DEBUG - ci_test]: (api_request)  : [Collection.flush] args: [], kwargs: {'timeout': 180} (api_request.py:62)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:58:43 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:58:43 - DEBUG - ci_test]: (api_request)  : [Collection.flush] args: [], kwargs: {'timeout': 180} (api_request.py:62)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:58:44 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:58:44 - INFO - ci_test]: assert flush: 3.0350630283355713, entities: 432421 (test_all_collections_after_chaos.py:67)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:58:44 - INFO - ci_test]: index info: [{'collection': 'Checker__vaAlMs3o', 'field': 'voice_emb', 'index_name': 'voice_emb', 'index_param': {'params': {'M': 48, 'efConstruction': 500}, 'index_type': 'HNSW', 'metric_type': 'L2'}}, {'collection': 'Checker__vaAlMs3o', 'field': 'int64', 'index_name': 'int64', 'index_param': {'index_type': 'INVERTED'}}, {'collection': 'Checker__vaAlMs3o', 'field': 'float', 'index_name': 'float', 'index_param': {'index_type': 'INVERTED'}}, {'collection': 'Checker__vaAlMs3o', 'field': 'varchar', 'index_name': 'varchar', 'index_param': {'index_type': 'INVERTED'}}, {'collection': 'Checker__vaAlMs3o', 'field': 'float_vector', 'index_name': 'float_vector', 'index_param': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 48, 'efConstruction': 500}}}, {'collection': 'Checker__vaAlMs3o', 'field': 'image_emb', 'index_name': 'image_emb', 'index_param': {'params': {'M': 48, 'efConstruction': 500}, 'index_type': 'HNSW', 'metric_type': 'L2'}}, {'collection': 'Checker__vaAlMs3o', 'field': 'text_emb', 'index_name': 'text_emb', 'index_param': {'params': {'M': 48, 'efConstruction': 500}, 'index_type': 'HNSW', 'metric_type': 'L2'}}] (test_all_collections_after_chaos.py:71)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:58:44 - INFO - ci_test]: index info: [{'collection': 'Checker__vaAlMs3o', 'field': 'float', 'index_name': 'float', 'index_param': {'index_type': 'INVERTED'}}, {'collection': 'Checker__vaAlMs3o', 'field': 'varchar', 'index_name': 'varchar', 'index_param': {'index_type': 'INVERTED'}}, {'collection': 'Checker__vaAlMs3o', 'field': 'float_vector', 'index_name': 'float_vector', 'index_param': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 48, 'efConstruction': 500}}}, {'collection': 'Checker__vaAlMs3o', 'field': 'image_emb', 'index_name': 'image_emb', 'index_param': {'params': {'M': 48, 'efConstruction': 500}, 'index_type': 'HNSW', 'metric_type': 'L2'}}, {'collection': 'Checker__vaAlMs3o', 'field': 'text_emb', 'index_name': 'text_emb', 'index_param': {'params': {'M': 48, 'efConstruction': 500}, 'index_type': 'HNSW', 'metric_type': 'L2'}}, {'collection': 'Checker__vaAlMs3o', 'field': 'voice_emb', 'index_name': 'voice_emb', 'index_param': {'params': {'M': 48, 'efConstruction': 500}, 'index_type': 'HNSW', 'metric_type': 'L2'}}, {'collection': 'Checker__vaAlMs3o', 'field': 'int64', 'index_name': 'int64', 'index_param': {'index_type': 'INVERTED'}}] (test_all_collections_after_chaos.py:86)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:58:44 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:58:44 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:58:44 - DEBUG - ci_test]: (api_request)  : [Collection.search] args: [[[0.11189364149366642, 0.10804722748379257, 0.0989093801059103, 0.13206884939958294, 0.14590195419386068, 0.020602707455921295, 0.05733112849772217, 0.13761296342479853, 0.11975534963558043, 0.09231903039298664, 0.10587827504776047, 0.09323391842476704, 0.14003310436128355, 0.013613231996936141, 0......., kwargs: {} (api_request.py:62)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:58:44 - DEBUG - ci_test]: (api_response) : ['[]']  (api_request.py:37)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:58:44 - INFO - ci_test]: assert search: 0.008784055709838867 (test_all_collections_after_chaos.py:99)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:58:44 - DEBUG - ci_test]: (api_request)  : [Collection.query] args: ['int64 in [-3000, -2999, -2998, -2997, -2996, -2995, -2994, -2993, -2992, -2991, -2990, -2989, -2988, -2987, -2986, -2985, -2984, -2983, -2982, -2981, -2980, -2979, -2978, -2977, -2976, -2975, -2974, -2973, -2972, -2971, -2970, -2969, -2968, -2967, -2966, -2965, -2964, -2963, -2962, -2961, -2960, -......, kwargs: {} (api_request.py:62)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:58:44 - DEBUG - ci_test]: (api_response) : []  (api_request.py:37)

[2024-04-18T18:03:13.130Z] [2024-04-18 17:58:44 - INFO - ci_test]: assert query result 0: 0.035094261169433594 (test_all_collections_after_chaos.py:106)
```

### Expected Behavior

the data can be queried

### Milvus Log

failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test-cron/detail/chaos-test-cron/13488/pipeline
log:
[artifacts-querynode-pod-kill-13488-server-logs.tar.gz](https://github.com/milvus-io/milvus/files/15032537/artifacts-querynode-pod-kill-13488-server-logs.tar.gz)

**Comment by zhuwenxing** [comment]
it can stable reproduced, so set it as critical

**Comment by zhuwenxing** [comment]
/assign
verified and fixed in `master-20240429-ac82cef0-amd64`
